consultation_analyser/consultations/views/answers.py

```diff
@@ -18,150 +19,285 @@ class DataDict(TypedDict):
+    return cache.get(cache_key)
+
+
+def set_cached_theme_summary(question_id: str, filters_hash: str, data: list[dict], timeout: int = 300):
+    """Cache theme summary for 5 minutes"""
+    cache_key = f"theme_summary:{question_id}:{filters_hash}"
+    cache.set(cache_key, data, timeout)
+
+
+def get_filters_hash(filters: FilterParams) -> str:
+    """Generate a hash for caching based on filter parameters"""
+    return hashlib.md5(str(sorted(filters.items())).encode()).hexdigest()
+
+
+def build_respondent_data(respondent: models.Respondent, response: models.Response) -> dict:
+    """Extract respondent data building to separate function"""
+    data = {
+        "id": f"response-{respondent.identifier}",
+        "identifier": str(respondent.identifier),
+        "sentiment_position": "",
+        "free_text_answer_text": response.free_text or "",
+        "demographic_data": respondent.demographics or {},
+        "themes": [],
+        "multiple_choice_answer": [response.chosen_options] if response.chosen_options else [],
+        "evidenceRich": False,
+        "individual": respondent.demographics.get("individual", False),
+    }
+    
+    if hasattr(response, "annotation") and response.annotation:
+        annotation = response.annotation
+        
+        if annotation.sentiment:
+            data["sentiment_position"] = annotation.sentiment
+        
+        if annotation.evidence_rich == models.ResponseAnnotation.EvidenceRich.YES:
+            data["evidenceRich"] = True
+        
+        # Add themes (already prefetched)
+        data["themes"] = [
+            {
+                "id": theme.id,
+                "stance": None,  # Stance is no longer stored in new models
+                "name": theme.name,
+                "description": theme.description,
+            }
+            for theme in annotation.themes.all()
+        ]
+    
+    return data
+
+
+def get_demographic_options(consultation: models.Consultation) -> dict[str, list]:
+    """Discover all demographic fields and their possible values in a consultation"""
+    # Get all unique demographic keys and values
+    demographic_data = {}
+    
+    # Query all respondents for this consultation
+    respondents = models.Respondent.objects.filter(consultation=consultation)
+    
+    for respondent in respondents:
+        if respondent.demographics:
+            for key, value in respondent.demographics.items():
+                if key not in demographic_data:
+                    demographic_data[key] = set()
+                # Convert value to string for consistency
+                demographic_data[key].add(str(value))
+    
+    # Convert sets to sorted lists
+    return {key: sorted(list(values)) for key, values in demographic_data.items()}
 
-        # Update cache
-        cache.set(cache_key, respondents, timeout=cache_timeout)
-    return respondents
+
+def derive_option_summary_from_responses(responses) -> list[dict]:
```

Comment:
Could this be slow when we have lots of responses?

I think we could make it faster maybe by getting all the options from the question, and then doing counts?

Something like:
```
def count_responses_for_all_options(question):
    """Count how many responses contain each option for a specific question."""
    option_counts = Counter()

    for option in question.multiple_choice_options:
        # Filter responses that contain the specific option
        count = Response.objects.filter(question=question, chosen_options__contains=[option]).count()
        option_counts[option] = count

```
Untested and untimed!

Perhaps one to look at once we have multiple choice responses to work with.

Reply:
Do we have somewhere to store this info for what we might want to change in future? Should we store it somewhere? (Like a Trello ticket?) Especially as it might not be us doing it.